inst/htmlwidgets/daterangepicker-bindings.js

```diff
@@ -168,57 +168,48 @@ $.extend(DateRangePickerBinding, {
       $("#"+data.id).attr("style", data.style);
     }
     // Update ranges
-    if (data.hasOwnProperty("ranges")) {
-      if (data.ranges !== undefined || data.ranges !== null) {
-        for (var key in data.ranges) {
-      		if (typeof data.ranges[key] === "string") {
-      		  data.ranges[key] = [moment(data.ranges[key]), moment(data.ranges[key])];
-      		} else {
-      		  for (var i in [0,1]) {
-      		    data.ranges[key].push(moment(data.ranges[key][0]));
-      		    data.ranges[key].shift();
-      		  }
-      		}
+    if (data.hasOwnProperty("ranges") && data.hasOwnProperty("rangeNames")) {
+      var ranges = {}
+      if (data.ranges !== undefined && data.ranges !== null &&
+          data.rangeNames !== undefined && data.rangeNames !== null) {
+        for (var i in data.ranges) {
+          if (typeof data.ranges[i] === "string") {
+            ranges[data.rangeNames[i]] = [moment(data.ranges[i]), moment(data.ranges[i])];
+          } else {
+            ranges[data.rangeNames[i]] = [moment(data.ranges[i][0]), moment(data.ranges[i][1])];
+          }
         }
       }
-      var ranges = {}
-      for (range in data.ranges) {
-          if (typeof data.ranges[range][0] === 'string')
-              start = moment(data.ranges[range][0]);
-          else
-              start = moment(data.ranges[range][0]);
+      var pickerRanges = {}
+      for (range in ranges) {
+        start = ranges[range][0];
+        end = ranges[range][1];
 
-          if (typeof data.ranges[range][1] === 'string')
-              end = moment(data.ranges[range][1]);
-          else
-              end = moment(data.ranges[range][1]);
```

Comment:
This was redundant, since `moment()` is  already called  in the block before this one.